diff --git a/drivers/mmc/host/sdhci-of-xilinxps.c b/drivers/mmc/host/sdhci-of-xilinxps.c
index f79e0db..377b5d8 100644
--- a/drivers/mmc/host/sdhci-of-xilinxps.c
+++ b/drivers/mmc/host/sdhci-of-xilinxps.c
@@ -44,9 +44,12 @@ static unsigned int zynq_of_get_max_clock(struct sdhci_host *host)
 
 	return pltfm_host->clock;
 }
-
+static unsigned int zynq_of_get_ro(struct sdhci_host *host){
+  return 0;
+}
 static struct sdhci_ops sdhci_zynq_ops = {
 	.get_max_clock = zynq_of_get_max_clock,
+	.get_ro = zynq_of_get_ro,
 };
 
 static struct sdhci_pltfm_data sdhci_zynq_pdata = {
diff --git a/drivers/net/ethernet/xilinx/xilinx_emacps.c b/drivers/net/ethernet/xilinx/xilinx_emacps.c
index 683b20b..92b1a2c 100644
--- a/drivers/net/ethernet/xilinx/xilinx_emacps.c
+++ b/drivers/net/ethernet/xilinx/xilinx_emacps.c
@@ -1835,7 +1835,7 @@ static int xemacps_open(struct net_device *ndev)
 {
 	struct net_local *lp = netdev_priv(ndev);
 	int rc;
-
+	int reg;
 	dev_dbg(&lp->pdev->dev, "open\n");
 	if (!is_valid_ether_addr(ndev->dev_addr))
 		return  -EADDRNOTAVAIL;
@@ -1852,7 +1852,12 @@ static int xemacps_open(struct net_device *ndev)
 		dev_err(&lp->pdev->dev, "pm_runtime_get() failed, rc %d\n", rc);
 		goto err_free_rings;
 	}
-
+	reg = xemacps_read(lp->baseaddr, XEMACPS_MODID_OFFSET);
+	while(reg == 0){
+	  printk("Waiting for xemacps to be ready\n");
+	  udelay(1);
+	  reg = xemacps_read(lp->baseaddr, XEMACPS_MODID_OFFSET);
+	}
 	xemacps_init_hw(lp);
 	napi_enable(&lp->napi);
 	rc = xemacps_mii_probe(ndev);
